server/src/routes/api/conversations.js

```diff
@@ -0,0 +1,118 @@
+const router = require("express").Router();
+const {
+  User,
+  Conversation,
+  Message,
+  LastSeen,
+  LastMessage
+} = require("../../db/models");
+const { Op } = require("sequelize");
+const onlineUsers = require("../../onlineUsers");
+
+const fetchConversations = async (userId) => {
+  return Conversation.findAll({
+    where: {
+      [Op.or]: {
+        user1Id: userId,
+        user2Id: userId,
+      },
+    },
+    attributes: ["id"],
+    order: [[Message, "createdAt", "ASC"]],
+    include: [
+      { model: Message, order: ["createdAt", "ASC"] },
+      {
+        model: LastSeen,
+        where: {
+          userId: {
+            [Op.not]: userId,
+          }
+        },
+        attributes: ["messageId"],
+        required: false
+      },
+      {
+        model: LastMessage,
+        where: {
+          userId: {
+            [Op.not]: userId,
+          }
+        },
+        attributes: ["messageId"],
+        required: false
+      },
+      {
+        model: User,
+        as: "user1",
+        where: {
+          id: {
+            [Op.not]: userId,
+          },
+        },
+        attributes: ["id", "username", "photoUrl"],
+        required: false,
+      },
+      {
+        model: User,
+        as: "user2",
+        where: {
+          id: {
+            [Op.not]: userId,
+          },
+        },
+        attributes: ["id", "username", "photoUrl"],
+        required: false,
+      },
+    ],
+  });
+};
+
+// get all conversations for a user, include latest message text for preview, and all messages
+// include other user model so we have info on username/profile pic (don't include current user info)
+// TODO: for scalability, implement lazy loading
+router.get("/", async (req, res, next) => {
+  try {
+    if (!req.user) {
+      return res.sendStatus(401);
+    }
+    const userId = req.user.id;
+    const conversations = await fetchConversations(userId);
+
+    for (let i = 0; i < conversations.length; i++) {
+      const convo = conversations[i];
+      const convoJSON = convo.toJSON();
```

Comment:
Is this correct? The `.toJSON()` the use of this method is to convert [Dates to JSON](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON).

Reply:
it is part of the objects returned by sequelize